extensions/wikia/AdEngine/js/lookup/lookupFactory.js

```diff
@@ -123,14 +122,23 @@ define('ext.wikia.adEngine.lookup.lookupFactory', [
 			});
 		}
 
-		function waitForResponse(milisToTimeout) {
-			return Promise.createWithTimeout(function (resolve) {
-				if (hasResponse()) {
-					resolve();
-				} else {
-					addResponseListener(resolve);
-				}
-			}, milisToTimeout);
+		function waitForResponse(onSuccess, onTimeout, millisToTimeout) {
+			var resolved = false;
+
+			if (hasResponse()) {
+				onSuccess();
+			} else {
+				setTimeout(function () {
+					onTimeout();
```

Comment:
I think function `onTimeout` will be launched every time when `hasResponse()` will be `false` even after correct `onSuccess` run from line 138 (https://github.com/Wikia/app/pull/14914/files#diff-b2d2d5acf2ecf54646c8cb7de3bf3185R138)